plugins/richcombo/plugin.js

```diff
@@ -399,6 +404,37 @@ CKEDITOR.plugins.add( 'richcombo', {
 					listener.removeListener();
 				} );
 				this._.listeners = [];
+			},
+
+			/**
+			 * Selects richcombo item based on the first matching result from the given filter function.
+			 * Filter function takes an object as an argumnet with `value` and `text` fields. Values of those
+			 * fields match to argument passed in `CKEDITOR.ui.richcombo.add` method.
+			 *
+			 * @since 4.14.0
+			 * @param {Function} callback function should return `true` if found matching element
+			 * @param {*} callback.item object containing value and text fields which are compared by this callback
+			 */
+			select: function( callback ) {
+				if ( CKEDITOR.tools.isEmpty( this._.items ) ) {
+					this.createPanel( this._.editor );
+				}
+
+				this.select = function( callback ) {
+					var value;
+
+					for ( value in this._.items ) {
+						if ( callback( {
+							value: value,
+							text: this._.items[ value ]
+						} ) ) {
+							this.setValue( value );
+							return;
+						}
+					}
+				};
+
+				this.select( callback );
```

Comment:
It seems like it could be simplified when you drop panel creation into simple for loop (inner implementation of `this.select` method).